src/plugins/visualizations/public/embeddable/visualize_embeddable.ts

```diff
@@ -465,4 +501,87 @@ export class VisualizeEmbeddable
       { showSaveModal: true, saveModalTitle }
     );
   };
+
+  /**
+   * Fetches any VisLayers, and filters out to only include ones in the list of
+   * input resource IDs, if specified. Assigns them to this.visLayers.
+   * Note this fn is public so we can fetch vislayers on demand when needed,
+   * e.g., generating other vis embeddables in the view events flyout.
+   */
+  public async populateVisLayers(): Promise<void> {
+    const visLayers = await this.fetchVisLayers();
+    this.visLayers =
+      this.visAugmenterConfig?.visLayerResourceIds === undefined
+        ? visLayers
+        : visLayers.filter((visLayer) =>
+            this.visAugmenterConfig?.visLayerResourceIds?.includes(visLayer.pluginResource.id)
+          );
+  }
+
+  /**
+   * Collects any VisLayers from plugin expressions functions
+   * by fetching all AugmentVisSavedObjects that match the vis
+   * saved object ID.
+   */
+  fetchVisLayers = async (): Promise<VisLayers> => {
```

Comment:
I tend to agree with the automation that thinks this fn should have unit tests. Can we add a follow-up task to write some tests for this to better define the intended behavior?

Reply:
I'm afraid of the extreme complexity and mocking that would need to be done to try to include this in unit tests, versus functional/cypress tests. The reason being, it involves integration with an external plugin that has a registered expression function. Some of the things that would need to be mocked are:
- mocked time range / query / filters from `data plugin`
- `augment-vis` saved obj loader responses
- `UISettings` service to see if the feature is enabled
- `expressions` plugin to execute the expression
- a mock expressions function that returns a mock `VisLayer`
- a mock plugin somehow that has registered an expressions function
- saved object loader response to delete other saved objects if doing cleanup
- potentially `notifications` service

Most of the individual functions within this one are covered via unit tests. Do you think it's necessary to try to set up UT to test this function as a whole, or leave it to cypress?